### DISCUSSION TITLE
Recommended way to force laravel to load env earlier

### DISCUSSION THREAD

**Discussion by lk77** [laravel/framework#47158] [Q&A] [answered]
Hello,

i'm in a situation where i need an environement variable, but when i need it, inside prepare function of DuskTestCase, the environement is not yet available, it's only loaded when setUp is called.

So i need laravel to load the environement, 

i'm doing that :

````
app(LoadEnvironmentVariables::class)->bootstrap(app());
````

is it the best way / recommended way ? 

thanks.

**Comment by crynobone** [comment, marked as answer]
For such requirements I would just read .env directly using DotEnv classes.

**Reply by lk77** [reply]
That's what i was doing earlier with `$env = \Dotenv\Dotenv::createArrayBacked('./')->load();` but i was searching for a solution allowing me to use env() helper instead, i guess i will keep using LoadEnvironmentVariables for now
